geps/gep-3567/index.md

```diff
@@ -0,0 +1,158 @@
+spec:
+  gatewayClassName: example
+  listeners:
+  - name: foo-https
+    protocol: HTTPS
+    port: 443
+    hostname: foo.example.com
+    tls:
+      certificateRefs:
+      - kind: Secret
+        group: ""
+        name: foo-example-com-cert  # SAN: foo.example.com
+  - name: wildcard-https
+    protocol: HTTPS
+    port: 443
+    hostname: "*.example.com"
+    tls:
+      certificateRefs:
+      - kind: Secret
+        group: ""
+        name: wildcard-example-com-cert  # SAN: *.example.com
+```
+
+
+The Gateway API definition requires requests to `foo.example.com` to be
+associated with the `foo-https` listener, on connections negotiated with
+`foo-example-com-cert`.
+
+Suppose a client sends a request to `bar.example.com`, specifying that as the
+SNI hostname, and establishes a TLS connection attached to the `wildcard-https`
+Listener. And then it sends a subsequent request to `foo.example.com`. The
+client can correctly reuse its existing TLS connection for the second request,
+because the `wildcard-example-com-cert` is valid also for `foo.example.com`. But
+now the Gateway has a problem: Routing the request via the `wildcard-https`
+Listener violates the intent of the configuration, and routing via the
+`foo-https` Listener is inconsistent with the connection’s having been
+negotiated with the other Listener’s certificate.
+
+Mapping a request to a Listener matters if the Gateway configuration has
+different
+[HTTPRoutes](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.HTTPRoute)
+bound to the different Listeners. It also matters if the Listeners have
+different
+[GatewayTlsConfigs](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.GatewayTLSConfig)
+attached, for example if one Listener uses mutual TLS and the other does not.
+
+
+### Interaction with Client Cert Validation
+
+[GEP-91](https://gateway-api.sigs.k8s.io/geps/gep-91/) introduced Client
+Certificate Validation to Gateway Listeners as a new experimental concept. If an
+implementation is unable to properly isolate HTTPS listeners, this could result
+in this Client Cert Validation being bypassed. Before this feature can graduate
+beyond experimental, we’ll need to resolve this underlying issue.
+
+## Proposal
+
+### A) Add Warning in Gateway Status
+A new condition will be added to Gateways: `OverlappingTLSConfig`.
+Implementations MUST add this condition to status when a Gateway is configured
+with TLS configuration across multiple Listeners. Implementations MAY add this
+condition to status when a Gateway is configured with overlapping TLS
+certifications.
+
+### B) Modify API Spec to recommend sending 421s
+The Gateway spec for `listener.hostname` will be updated to recommend returning
+a 421 when this problem occurs.
+
+### C) Top Level Gateway TLS Config for Client Cert Validation
+
+GEP-91 will be updated to move this configuration to a new top level Gateway TLS
+config instead of per-listener.
+
```

Comment:
do you mean `move` or `copy` the client cert validation field ? imo there's still value in per listener client validation for non intersecting hostnames

Reply:
I'd argue that it should be "move", but the only thing I'm trying to communicate here is that we'll need follow up changes in GEP-91. Will update the wording to be more open-ended here.

Reply:
Loosened this to be more open-ended and more clearly a follow up item.